### ISSUE TITLE
Document the relation between hook levels and logrus level.

### ISSUE THREAD

**Issue by aaronkavlie-wf** [Sirupsen/logrus#334]
It's often unclear to users (spoken from experience as a hook author) whether or not the logrus logging level (as set with a call to `logrus.SetLevel()`) also affects the minimum logging level that hooks will output. The waters are muddied by the fact that the the `Hook` interface has a `Levels()` method, and hooks [sometimes][1] [accept][2] an argument to define a level list of minimum threshold. This gives the appearance that the hook will operate independent of the logging level set for logrus's standard output.

I couldn't find anything that clarifies this in the README or code comments.

[1]: https://github.com/evalphobia/logrus_sentry
[2]: https://github.com/nubo/hiprus

**Comment by travisby** [comment]
I had just spent the last hour or so attempting to create a hook that would log when the logger's level would not -- an assumption that this issue wants to address.

Would it be acceptable to make hook levels and the logger levels independent of eachother?  Or is that too much of a breaking change?

**Comment by aaronkavlie-wf** [comment]
Yeah, in a lot of use cases (including mine) you just want the hook to reflect the level set by logrus without thinking about it.

Current implementation aside, this would be ideal:

- Hooks inherit the logrus minimum level by default
- *But* it's possible to explicitly adjust the level to be higher/lower.

This would perhaps eliminate the ability to send on e.g. `logrus.ErrorLevel` only -- which is possible as `Levels()` returns a slice of levels. But I don't know why you would want to include errors, while excluding fatal & panic levels. Every example I've looked at includes all levels more severe than their chosen threshold.

**Comment by aybabtme** [comment]
@aaronkavlie-wf thanks for the idea. I haven't faced this issue myself, so I'm not too familiar with how I'd go about explaining this. If you could submit a PR with documentation changes, I'll be happy to have a look.
